### DISCUSSION TITLE
Is Transport security with CustomUserNamePasswordValidator supported?

### DISCUSSION THREAD

**Discussion by mvarblow** [CoreWCF/CoreWCF#544] [Q&A]
All the authentication samples seem to use a security mode of TransportWithMessageCredential. I'm trying to use plain Transport with a client credential type of Basic and a CustomUserNamePasswordValidator. But this setup doesn't seem to have any effect. Not only do anonymous requests flow through, the UserNamePasswordValidator.ValidateAsync method is never called. Am I configuring this correctly? Are Transport credentials not supported yet?

        services.AddSingleton<IServiceBehavior>(_ => new ServiceCredentials()
        {
            UserNameAuthentication =
            {
                UserNamePasswordValidationMode = CoreWCF.Security.UserNamePasswordValidationMode.Custom,
                CustomUserNamePasswordValidator = new MyUserNamePasswordValidator()
            }
        });
        services.AddTransient<AccessAgreementService>();

        Binding soap11Binding = new BasicHttpBinding()
        {
            Security = new BasicHttpSecurity
            {
                Mode = BasicHttpSecurityMode.Transport,
                Transport = new HttpTransportSecurity
                {
                    ClientCredentialType = HttpClientCredentialType.Basic
                }
            }
        }
        app.UseServiceModel(builder =>
        {
            builder.AddService<AccessAgreementService>()
                    .AddServiceEndpoint<AccessAgreementService, IAccessAgreementService>(soap11Binding, "/AccessAgreement.asmx");
        });

FWIW, based on the old WCF doco, this _looks_ like the right way to set things up. But if I'm missing something, I'd appreciate a point in the right direction. Thanks!

https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/how-to-use-a-custom-user-name-and-password-validator

**Comment by birojnayak** [comment]
this should help https://github.com/CoreWCF/CoreWCF/blob/main/src/CoreWCF.Http/tests/BasicHttpTransportMessage.cs#L142 or this should https://github.com/CoreWCF/CoreWCF/blob/main/src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs#L109 ... it's supported...
